test/test.c

```diff
@@ -3928,301 +3929,271 @@ static hb_dict_t * PreparePreset(const char *preset_name)
                     MAX(hb_str_vlen(acodecs),
                         hb_str_vlen(anames)))))))))));
 
-        hb_dict_t *audio_dict, *last_audio_dict = NULL;
-        // Add audio dict entries to list if needed
-        for (ii = 0; ii < count; ii++)
+        if (list_len < count)
         {
-            audio_dict = hb_value_array_get(list, ii);
-            if (audio_dict == NULL)
+            // More command line settings specified than preset currently supports.
+            // Duplicate the last audio encoder settings in the preset or create a
+            // new one if the preset audio list is empty.
+            hb_dict_t *audio_dict_stub;
+            if (list_len == 0)
             {
-                break;
+                audio_dict_stub = hb_dict_init();
             }
-            last_audio_dict = audio_dict;
-        }
-        // More settings specified than preset currently supports.
-        // Duplicate the last audio encoder settings in the preset.
-        for (; ii < count && last_audio_dict != NULL; ii++)
-        {
-            audio_dict = hb_value_dup(last_audio_dict);
-            hb_value_array_append(list, audio_dict);
+            else
+            {
+                audio_dict_stub = hb_value_dup(hb_value_array_get(list, list_len - 1));
+            }
+
+            // The command line allows a shortcut where the final audio
+            // setting listed in the comma separated list gets
+            // replicated and applied to any remaining entries.
+            // If the setting is not specified at all or the comma separated
+            // list end in a `,`, defaults are used.
+            // some_setting[last][0] == 0 indicates the list ended in `,`
+            int last = hb_str_vlen(acodecs) - 1;
+            if (last >= 0 && acodecs[last][0] != 0)
+            {
+                hb_dict_set(audio_dict_stub, "AudioEncoder",
+                             hb_value_string(acodecs[last]));
+            }
+            else
+            {
+                const char *enc;
+                enc = hb_audio_encoder_get_short_name(
+                        hb_audio_encoder_get_default(mux));
+                hb_dict_set(audio_dict_stub, "AudioEncoder",
+                        hb_value_string(enc));
+            }
+            last = hb_str_vlen(abitrates) - 1;
+            int last_q = hb_str_vlen(aqualities) - 1;
+            if (last_q > last && aqualities[last_q][0] != 0)
+            {
+                hb_dict_set(audio_dict_stub, "AudioTrackQualityEnable",
+                            hb_value_bool(1));
+                hb_dict_set(audio_dict_stub, "AudioTrackQuality",
+                    hb_value_double(strtod(aqualities[last_q], NULL)));
+            }
+            else if (last >= 0 && abitrates[last][0] != 0)
+            {
+                hb_dict_set(audio_dict_stub, "AudioBitrate",
+                    hb_value_int(atoi(abitrates[last])));
+            }
+            else
```

Comment:
This `else` means the bitrate is always overridden to the default value, I think there should be another check to see if the `audio_dict_stub` already has the `AudioBitrate` key or not, and only set the default bitrate when it doesn't

Reply:
Oh, right. I actually meant to do that and forgot. Same should happen with encoder and that fixes part of the issue you brought up above.